americangut/power_plots.py

```diff
@@ -0,0 +1,581 @@
+        else:
+            effect_means[idp] = np.nanmean(eff, None)
+            effect_bounds[idp] = confidence_bound(eff, alpha, None)
+
+    return effect_means, effect_bounds
+
+
+def summarize_effect(order, fecal_cats, a_eff_means, b_eff_means, b_eff_bounds,
+                     a_eff_bounds):
+    """Creates a pretty html formatted table of the results
+
+    The output table can be displayed in an IPython notebook as an HTML object,
+    rather than building a pandas table. Hopefully, this will allow for a
+    beautiful display. The numeric cells are split into three columns to allow
+    for numerical alignment.
+
+    The effect sizes are rounded to 5 samples, and the standard devation are
+    rounded to 1 sample.
+
+    Parameters
+    ----------
+    order : array
+        The order in which categories should be displayed.
+    fecal_cats : list
+        A list giving the fecal category as a formatted string
+    a_eff_means, b_eff_means : list of arrays
+        An array with the mean effects. The output of `collate_effect_size`.
+    a_eff_bounds, b_eff_bounds : list of arrays
+         An array with the mean bound. The confidence interval for the mean
+         is given by [eff_mean - eff_bound, eff_mean + eff_bound].
+         This is an output of `collate_effect_size`.
+
+    Returns
+    -------
+    table : str
+        An html-formmated table which can be rendered in IPython.
+
+
+    """
+    # Sets up the html for the header
+    table = ['<table style="border-style:hidden;',
+             '              border-collapse:collapse',
+             '              line-height:120%',
+             '              ">',
+             '\t<tr>',
+             '\t\t<th style="text-align:center;',
+             '\t\t           background-color:black;',
+             '\t\t           color:white',
+             '\t\t           ">',
+             '\t\t\tCategory'
+             '\t\t</th>',
+             '\t\t<th style="text-align:center;',
+             '\t\t           background-color:black;',
+             '\t\t           color:white";',
+             '\t\t    colspan=3>',
+             '\t\t\tAlpha',
+             '\t\t</th>',
+             '\t\t<td style="border-hidden;',
+             '\t\t           background-color:black;',
+             '\t\t           padding:20px">',
+             '\t\t<th style="text-align:center;',
+             '\t\t           background-color:black;',
+             '\t\t         color:white";',
+             '\t\t  colspan=3>',
+             '\t\t\tBeta',
+             '\t\t</th>',
+             '\t</tr>']
+    # Loops through each row
+    for idx in order:
+        # Gets the effect sizes
+        cat = fecal_cats[idx]
+        a_eff_m = a_eff_means[idx]
+        a_eff_b = a_eff_bounds[idx]
+        b_eff_m = b_eff_means[idx]
+        b_eff_b = b_eff_bounds[idx]
+        # Gets the fitted values
+        a_eff_fit = np.ceil(ft.solve_power(a_eff_m, nobs=None, power=0.8,
```

Comment:
why is this happening in a function that is responsible for producing a visual representation of the data? it is a very good idea to separate presentation from business logic...